packages/appsync-emulator-serverless/cloudFormationProcessor.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * This module provides tools to traverse a given object and apply transformations
+ * to the object using a cloud formation like syntax. We're not trying to replace
+ * all CF functionality just what is needed to run the appsync emulator.
+ */
+
+const DynamoDBTable = 'AWS::DynamoDB::Table';
+
+function humanObjectPath(objectPath) {
+  return objectPath.join('.');
+}
+
+function lookupResourcesFromCtx(resource, ctx) {
+  return ctx.resources && ctx.resources.Resources[resource];
+}
+
+function lookupResourcePropertyFromCtx(cfObject, prop) {
+  return cfObject && cfObject.Properties && cfObject.Properties[prop];
+}
+
+function lookupDynamodbTableName(cfObject, { dynamodbTables }) {
+  // support aliasing for test isolation.
+  const tableName = lookupResourcePropertyFromCtx(cfObject, 'TableName');
+  if (dynamodbTables[tableName]) {
+    return dynamodbTables[tableName];
+  }
+
+  // we do not require an alias and fallback to the name in resources if not found.
+  return tableName;
+}
+
+function cfRef(value, ctx, objectPath) {
+  const cfObject = lookupResourcesFromCtx(value, ctx);
+  if (!cfObject) {
+    throw new Error(
+      `Cannot find referenced [Ref] cloud formation resource (${humanObjectPath(
+        objectPath,
+      )})`,
+    );
+  }
+
+  const { Type: type } = cfObject;
+  if (!type) {
+    // TODO: Add path
+    throw new Error(
+      `Unknown cloud formation object reference or type (${humanObjectPath(
+        objectPath,
+      )})`,
+    );
+  }
+
+  switch (type) {
+    case DynamoDBTable:
+      return lookupDynamodbTableName(cfObject, ctx);
+    default:
+      throw new Error(
+        `Unable to find logical ref resource for : ${type} (${humanObjectPath(
+          objectPath,
+        )})`,
+      );
+  }
+}
+
+const cloudFormationHandlers = {
+  Ref: cfRef,
+};
+
+function processObject(object, ctx, objectPath = []) {
```

Comment:
## Object path

I took extra care to make this debuggable... One of the key features of the emulator should be better iteration loop which means better error messages. This will allow one to track down which line of their cloud formation function is on.

When something goes wrong you will get an error message like `... (custom.appSync.bla)`